### ISSUE TITLE
AddressSanitizer SEGV src/array.c:259:20

### ISSUE THREAD

**Issue by revskills** [mruby/mruby#3244]
Repro file:

```sh
Base64: ICAqYSA9DTw8LTU1LDIsKmEgPS01PDwtNTUKd2hl9iAqYyA9AAA=
```
stacktrace
```sh
ASAN:DEADLYSIGNAL
=================================================================
==10068==ERROR: AddressSanitizer: SEGV on unknown address 0x00009fff8002 (pc 0x0000006068e9 bp 0x7fff2a2821d0 sp 0x7fff2a282150 T0)
==10068==The signal is caused by a READ memory access.
    #0 0x6068e8 in ary_concat /home/fuzzer/git/mruby/src/array.c:259:20
    #1 0x606850 in mrb_ary_concat /home/fuzzer/git/mruby/src/array.c:273:3
    #2 0x5b281a in mrb_vm_exec /home/fuzzer/git/mruby/src/vm.c:2137:7
    #3 0x592b6a in mrb_vm_run /home/fuzzer/git/mruby/src/vm.c:766:10
    #4 0x5bd377 in mrb_top_run /home/fuzzer/git/mruby/src/vm.c:2452:12
    #5 0x64bde8 in load_exec /home/fuzzer/git/mruby/mrbgems/mruby-compiler/core/parse.y:5711:7
    #6 0x64a775 in mrb_load_file_cxt /home/fuzzer/git/mruby/mrbgems/mruby-compiler/core/parse.y:5720:10
    #7 0x4f28c7 in main /home/fuzzer/git/mruby/mrbgems/mruby-bin-mruby/tools/mruby/mruby.c:226:9
    #8 0x7f4281cbf82f in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x2082f)
    #9 0x41a0e8 in _start (/home/fuzzer/git/mruby/build/test/bin/mruby+0x41a0e8)

AddressSanitizer can not provide additional info.
SUMMARY: AddressSanitizer: SEGV /home/fuzzer/git/mruby/src/array.c:259:20 in ary_concat
```

**Comment by matz** [comment]
I confirmed the issue. Thank you for the report.

**Comment by matz** [comment]
fixed by 45aadd7
